### PR TITLE
[tune] Disallow setting resources_per_trial when it is already configured

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Raise an error if the user tries to set resources_per_trial in RLlib.

## Related issue number

https://groups.google.com/forum/#!topic/ray-dev/coaz8dgHyYw

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/tune/trial.py [hunk 1]
```diff
@@ -275,9 +278,22 @@ def __init__(self,
...
+            self.resources = None
+        if self.resources:
+            if resources:
+                raise ValueError(
+                    "Resources for {} have been automatically set to {} "
+                    "by its `default_resource_request()` method. Please "
+                    "clear the `resources_per_trial` option.".format(
+                        trainable_cls, self.resources))
+        else:
+            self.resources = Resources(cpu=1, gpu=0)
```
Comment (richardliaw) [inline python/ray/tune/trial.py]: doesn't this automatically set all resources to `Resources(cpu=1, gpu=0)`?

Comment (ericl) [reply python/ray/tune/trial.py]: Fixed.

**Comment by ericl** [comment]
Friendly ping on review.

File: python/ray/tune/trial.py [hunk 2]
```diff
@@ -275,9 +278,22 @@ def __init__(self,
...
+            self.resources = None
+        if self.resources:
+            if resources:
+                raise ValueError(
+                    "Resources for {} have been automatically set to {} "
+                    "by its `default_resource_request()` method. Please "
+                    "clear the `resources_per_trial` option.".format(
+                        trainable_cls, self.resources))
+        else:
+            self.resources = resources or Resources(cpu=1, gpu=0)
```
Comment (richardliaw) [inline python/ray/tune/trial.py]: hm can you make this a bit less convoluted?

```
        trainable_cls = self._get_trainable_cls()
        if trainable_cls and hasattr(trainable_cls,
                                     "default_resource_request"):
            default_resources = trainable_cls.default_resource_request(
                self.config)
            if default_resources and resources:
                raise ValueError(
                    "Resources for {} have been automatically set to {} "
                    "by its `default_resource_request()` method. Please "
                    "clear the `resources_per_trial` option.".format(
                        trainable_cls, default_resources))
            resources = default_resources
        self.resources = resources or Resources(cpu=1, gpu=0)
```

Comment (ericl) [reply python/ray/tune/trial.py]: Done

**Review by richardliaw — APPROVED** [review]

**Pull request merged**